main/src/ca/uwaterloo/flix/api/lsp/SymbolTag.scala

```diff
@@ -0,0 +1,16 @@
+package ca.uwaterloo.flix.api.lsp
+
+import org.json4s.JsonAST.{JInt, JValue}
+
+/**
+  * Represents a `SymbolTag` in LSP.
+  */
+trait SymbolTag {
+  def toJSON: JValue = this match {
+    case SymbolTag.Deprecated => JInt(1)
+  }
+}
+
+object SymbolTag {
+  case object Deprecated extends SymbolTag
```

Comment:
I am guessing there are more types, but you plan to add them later?

Reply:
I think this is the only one as for now: https://microsoft.github.io/language-server-protocol/specifications/specification-3-17/#symbolTag

However it's a pretty new feature, so probably other tags will be added in the future